src/Microsoft.ML.OnnxConverter/SaveOnnxCommand.cs

```diff
@@ -204,11 +208,39 @@ internal static void GetPipe(OnnxContextImpl ctx, IChannel ch, IDataView end, ou
                 var trueVariableName = ctx.AddIntermediateVariable(null, idataviewColumnName + ".output", true);
                 ctx.CreateNode("Identity", variableName, trueVariableName, ctx.GetNodeName("Identity"), "");
                 ctx.AddOutputVariable(outputData.Schema[i].Type, trueVariableName);
+
+                if (column.HasSlotNames())
+                    AddSlotNames(ctx, column);
             }
 
+            // Add metadata graph outputs
+
             return ctx.MakeModel();
         }
 
+        private static void AddSlotNames(OnnxContextImpl ctx, DataViewSchema.Column column)
+        {
+            VBuffer<ReadOnlyMemory<char>> slotNames = default;
+            column.GetSlotNames(ref slotNames);
+            IEnumerable<string> slotNamesAsStrings = slotNames.DenseValues().Select(name => name.ToString());
+
+            string opType = "LabelEncoder";
+            string labelEncoderInputName = $"mlnet.{column.Name}.unusedInput";
+            string labelEncoderOutputName = $"mlnet.{column.Name}.unusedOutput";
+            string labelEncoderNodeName = $"mlnet.{column.Name}.SlotNames";
+
+            string[] oneVals = new string[] { "one" };
+            long[] dims = new long[] { 1, 1 };
+            var one = ctx.AddInitializer(oneVals, dims, labelEncoderNodeName);
+
+            var labelEncoderOutput = ctx.AddIntermediateVariable(NumberDataViewType.Int64, labelEncoderOutputName, true);
+            var node = ctx.CreateNode(opType, one, labelEncoderOutput, labelEncoderNodeName);
+            node.AddAttribute("keys_strings", slotNamesAsStrings);
+            node.AddAttribute("values_int64s", Enumerable.Range(0, slotNames.Length).Select(x => (long)x));
```

Comment:
>values_int64s [](start = 31, length = 13)

Why do we need this?
Does it even work when slot names are not numbers?

Reply:
These are unused. But are specified only to satisfy ORT. 

---
In reply to: [381867237](https://github.com/dotnet/machinelearning/pull/4857#discussion_r381867237) [](ancestors = 381867237)